### PR TITLE
Bluetooth: audio: Fix various compilation warnings

### PR DESCRIPTION
**Pull request by MariuszSkamra**
Fix various compilation issues found in HAS and ASCS.

### CODE REVIEW

File: subsys/bluetooth/audio/Kconfig.has [hunk 1]
```diff
@@ -65,7 +65,10 @@ config BT_HAS_PRESET_COUNT
 	  This option sets the number of Hearing Access Service Presets
 	  that can be registered. Setting this value to 0 disables Presets support.
 
-if BT_HAS_PRESET_COUNT > 0
+config BT_HAS_PRESET_SUPPORT
+	def_bool BT_HAS_PRESET_COUNT > 0
```
Comment (Thalley) [inline subsys/bluetooth/audio/Kconfig.has:69]: The alternative here would be
```suggestion
	bool # hidden
	default y if BT_HAS_PRESET_COUNT > 0
```

Not sure which is preferred, but I guess either one works :)

File: subsys/bluetooth/shell/CMakeLists.txt [hunk 2]
```diff
@@ -70,7 +70,7 @@ zephyr_library_sources_ifdef(
   media_controller.c
   )
 zephyr_library_sources_ifdef(
-  CONFIG_BT_HAS
+  CONFIG_BT_HAS_PRESET_SUPPORT
```
Comment (Thalley) [inline subsys/bluetooth/shell/CMakeLists.txt:73]: This change doesn't look right: `HAS` can exist without preset support, right?

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/shell/CMakeLists.txt (hunk 2 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/shell/CMakeLists.txt:73]: This change is for shell support that has only preset related commands for now

File: subsys/bluetooth/audio/Kconfig.has (hunk 1 above)
Comment (MariuszSkamra) [reply subsys/bluetooth/audio/Kconfig.has:69]: def_bool is the one suggested on Zephyr website

File: subsys/bluetooth/shell/CMakeLists.txt (hunk 2 above)
Comment (Thalley) [reply subsys/bluetooth/shell/CMakeLists.txt:73]: Right, my bad. Didn't see that this if for the shell :)

**Review by Thalley — APPROVED** [review]

**Review by asbjornsabo — APPROVED** [review]

**Pull request merged**